### PR TITLE
docs: cleanup upgrade docs on main

### PR DESCRIPTION
**Pull request by marseel**
Accidentally, we left out 1.16/1.15 upgrade notes so let's remove it.
Leave Cilium CLI part though, as it's still relevant and also
cli-download documentation points to it.

### CODE REVIEW

File: Documentation/operations/upgrade.rst [hunk 1]
```diff
@@ -310,240 +310,15 @@ Annotations:
   add an additional ``--bpf-lb-external-control-plane=true`` to the command line, otherwise
   it is assumed that Kubernetes is present.
 
-.. _1.16_upgrade_notes:
-
-1.16 Upgrade Notes
```
Comment (joestringer) [inline Documentation/operations/upgrade.rst:315]: Could you update this to 1.17? with the same section headers as before?

Usually contributors need a hook of where to contribute their upgrade notes, so it helps to keep the structure alive early in the dev cycle.

**Review by joestringer — CHANGES_REQUESTED** [review]

File: Documentation/operations/upgrade.rst [hunk 2]
```diff
@@ -310,240 +310,15 @@ Annotations:
...
 
 Cilium CLI
 ~~~~~~~~~~
 
 Upgrade Cilium CLI to `v0.15.0 <https://github.com/cilium/cilium-cli/releases/tag/v0.15.0>`_
 or later to switch to `Helm installation mode <https://github.com/cilium/cilium-cli#helm-installation-mode>`_
-to install and manage Cilium v1.14. Classic installation mode is **not**
-supported with Cilium v1.14.
+to install and manage Cilium v1.17. Classic installation mode is **not**
+supported with Cilium v1.17.
```
Comment (joestringer) [inline Documentation/operations/upgrade.rst]: We should probably take a fresh look at this section and rewrite/remove it. Cilium 1.14 will be the last actively maintained version in a few short weeks, which will mean that we expect all users to be running with Helm installation mode.

File: Documentation/operations/upgrade.rst (hunk 1 above)
Comment (marseel) [reply Documentation/operations/upgrade.rst:315]: 1.17 was above it. I've added other sections though for easier development.

File: Documentation/operations/upgrade.rst (hunk 2 above)
Comment (marseel) [reply Documentation/operations/upgrade.rst]: yea, let me remove it. It seems like install/uninstall for classic mode was removed in v0.16.0 cli version already so it's not even really possible to uninstall with a newer cli versions

**Review by joestringer — APPROVED** [review]
LGTM thanks!

**Pull request merged**